clcache.py

```diff
@@ -164,33 +164,43 @@ def clean(self, stats, maximumSize):
         except ImportError:
             walker = os.walk
 
-        objects = [os.path.join(root, "object")
-                   for root, folder, files in walker(self.objectsDir)
-                   if "object" in files]
+        objectInfos = []
+        for x in xrange(16):
+            try:
+                objects = [os.path.join(root, "object")
+                       for root, folder, files in walker(self.objectsDir)
+                       if "object" in files]
+
+                objectInfos = [(os.stat(fn), fn) for fn in objects]
```

Comment:
if `stat` throws by deleting the cache, the `object` directory most likely has changed. but i don't have any strong opinion on this, so just skipping is fine as for me.

i had been using this patch for a while (when parallel processes were still an issue). these days, it is mainly that users can delete the cache without causing builds to fail.